Source/drlg_l1.cpp

```diff
@@ -517,58 +552,28 @@ void InitCryptPieces()
 
 void PlaceDoor(int x, int y)
 {
-	if ((L5dflags[x][y] & DLRG_PROTECTED) == 0) {
-		BYTE df = L5dflags[x][y] & 0x7F;
-		BYTE c = dungeon[x][y];
-
-		if (df == 1) {
-			if (y != 1 && c == 2)
-				dungeon[x][y] = 26;
-			if (y != 1 && c == 7)
-				dungeon[x][y] = 31;
-			if (y != 1 && c == 14)
-				dungeon[x][y] = 42;
-			if (y != 1 && c == 4)
-				dungeon[x][y] = 43;
-			if (x != 1 && c == 1)
-				dungeon[x][y] = 25;
-			if (x != 1 && c == 10)
-				dungeon[x][y] = 40;
-			if (x != 1 && c == 6)
-				dungeon[x][y] = 30;
-		}
-		if (df == 2) {
-			if (x != 1 && c == 1)
-				dungeon[x][y] = 25;
-			if (x != 1 && c == 6)
-				dungeon[x][y] = 30;
-			if (x != 1 && c == 10)
-				dungeon[x][y] = 40;
-			if (x != 1 && c == 4)
-				dungeon[x][y] = 41;
-			if (y != 1 && c == 2)
-				dungeon[x][y] = 26;
-			if (y != 1 && c == 14)
-				dungeon[x][y] = 42;
-			if (y != 1 && c == 7)
-				dungeon[x][y] = 31;
-		}
-		if (df == 3) {
-			if (x != 1 && y != 1 && c == 4)
-				dungeon[x][y] = 28;
-			if (x != 1 && c == 10)
-				dungeon[x][y] = 40;
-			if (y != 1 && c == 14)
-				dungeon[x][y] = 42;
-			if (y != 1 && c == 2)
-				dungeon[x][y] = 26;
-			if (x != 1 && c == 1)
-				dungeon[x][y] = 25;
-			if (y != 1 && c == 7)
-				dungeon[x][y] = 31;
-			if (x != 1 && c == 6)
-				dungeon[x][y] = 30;
-		}
+	auto tileId = static_cast<CatedralTile>(dungeon[x][y]);
+	if (L5dflags[x][y] == (DLRG_VDOOR | DLRG_HDOOR)) {
+		if (tileId == CatedralTile::DWall)
+			dungeon[x][y] = CatedralTile::HDoorVDoor;

```

Comment:
The only real difference I can see between this and the old logic is removing the checks for placing doors in the first row/column of the dungeon layer. Removing the checks seems fine since there's no valid way to have row 0 separated from row 2 that supports a door connection (that I can see anyway 😅 )  and there was no corresponding checks for the other near-edges, is this something that was ever triggered in dungeon generation?

Reply:
Also it checks for row/column 1 when applying the `DLRG_VDOOR | DLRG_HDOOR` flag, so it would never happen on row/column 1 :)

Reply:
After looking a bit closer I realize that none of the other cases can ever happen since it will always set the tile to h/vwall when adding the h/vdoor flag, and the reset of the code will avoid modifying it once the flag is set. So we can simply set the door tile directly when building the wall.